shared/actions/settings.tsx

```diff
@@ -347,22 +317,23 @@ function* refreshNotifications() {
     } || [])
 
   const groups = results.notifications
-  const notifications = mapValues(groups, group => ({
+  const notifications: {[key: string]: Types.NotificationsGroupState} = mapValues(groups, group => ({
     settings: group.settings.map(settingsToPayload),
     unsubscribedFromAll: group.unsub,
-  }))
+  })) as any // TODO fix
 
   yield Saga.put(
     SettingsGen.createNotificationsRefreshed({
-      // @ts-ignore
       notifications: I.Map(notifications),
     })
   )
 }
 
-const dbNuke = () => RPCTypes.ctlDbNukeRpcPromise()
+const dbNuke = async () => {
+  await RPCTypes.ctlDbNukeRpcPromise(undefined, Constants.settingsWaitingKey)
```

Comment:
Why `undefined` and not `{}`?

Reply:
you don't need to return anything from these functions. if you do, then you need to cover all paths (best `return false`). in this case it doesn't return anything. previously it returned whatever `RPCTypes.ctlDbNukeRpcPromise()` returned which is a little dangerous. we essentially never want to plumb that through